### ISSUE TITLE
dnscurve magic number

### ISSUE THREAD

**Issue by jedisct1** [chengr28/Pcap_DNSProxy#53]
Hello,

And congrats for your awesome Pcap_DNSProxy!

Just one thing: it looks like the client magic number for dnscrypt has to be specified in the configuration file.
This is not good. This number is not constant. It changes every time the certificate is updated.

The most frequently the certificate is rotated, the better it is for forward secrecy. Even though some public resolvers do not change the certificate very often because they are lazy or don't care about forward secrecy, the protocol recommends updating them [every 24 hours](https://github.com/jedisct1/dnscrypt-proxy/blob/master/DNSCRYPT-V2-PROTOCOL.txt#L415).

So, this should not be specified in the configuration file. The magic number to use is part of the certificate (probably `dnscurve_txt_signature.MagicNumber`) and has to be only read from here, since it can change anytime.

**Comment by chengr28** [comment]
Hi, I'm sorry for the delay.

When I add DNSCrypt protocol support, I found the standard process to get the encryption certificate which require to resolve a special domain without encryption. I think it's very danger in China mainland because of GFW, which can block responses or reply fake responses to make the standard process failed. So I designed a module call Encryption Only.

When Encryption Only module is on, this module needn't the standard process or any other data exchange without encryption but require fingerprint and magic numbers of server to resolve domains with encryption all time. So I write the fingerprint and magic number options into configuration file to support this module.

The Encryption Only module is off in default setting. Users can turn this module on in any time when they need this module. In next version, I will add a module which can read the configuration file immediately when the file is changed. Users can also change magic numbers in any time.

Thank you for your suggestion ^_^
